lib/WUI/nhttp/file_info.cpp

```diff
@@ -106,6 +106,7 @@ Step FileInfo::step(std::string_view, bool, uint8_t *output, size_t output_size)
             memcpy(output + written, LAST_CHUNK, last_chunk_len);
             written += last_chunk_len;
         }
+        dir.reset();
```

Comment:
Why do you need resetting the `dir` here? Just a precaution/saving stack space before creating the returned structures?

Reply:
Dir is (and always was) a member variable. It holds the position in listing file entries between sending individual files/packets. And I suspect there's some dynamic allocation (in `opendir`) that I don't know how to avoid :-(

It's not 100% necessary (it would get closed in the destructor soon), but I wanted to return the memory sooner if possible, as the integration tests often fail on out of memory (I'm not sure this helps, though).

Should I put some explanation into the code, or is this OK?

Reply:
explained enough :+1: